resotool/resoapp/tasks.py

```diff
@@ -0,0 +1,55 @@
+from celery import shared_task
+from celery.result import AsyncResult
+from django.core import mail
+
+from .models import Resolution, ResolutionEmail, SendStatus
+from resotool import settings
+
+
+def send_reso(reso_id, mail_text, subject, sender, celery=True):
+    resolution = Resolution.objects.get(pk=reso_id)
+    mail_list = []
+    mail_data_list = []
+    for recipient in resolution.addressee_set.all():
+        mail_data_list.append(
+            (subject, mail_text, sender, [recipient.__str__()])
+        )
+        reso_mail = ResolutionEmail(
+            addressee=recipient,
+            resolution=resolution,
+            email_text=mail_text,
+            status=SendStatus.IN_PROGRESS,
+        )
+        reso_mail.save()
+        mail_list.append(reso_mail)
+    send_results = mail.send_mass_mail(mail_data_list)
+
+    if type(send_results) is list:
+        for email, send_result in zip(mail_list, send_results):
+            send_result.then(on_reso_sent(email), on_send_error(email))
+        return send_results
+    else:
+        return send_results
+
+
+def on_reso_sent(email):
+    email = email
+
+    def return_func(result):
+        if result.get() == 1:
```

Comment:
`result.get()` returns the number of sent emails. This is based on the chunk size set in `resotool.settings.py`, which in our case is 1. So practically the only two expected results are 0 (if the was a problem while sending) and 1 (if the mail was sent). But if someone changes this setting, the return value could be anything from 0 to `CELERY_EMAIL_CHUNK_SIZE`.

At least in testing, this problem would also be caught in other places, since the number of return values gets compared to the number of mails that should have been sent, but I am not sure whether there would be anything else catching this error in production.

But then again, this is probably not the correct place to catch this error since a return value bigger than 1 would mean that there were still some mails, that were successfully sent. We just don't know which.

I think I will change this as requested and we should hope, no one changes that setting in the future.

Reply:
Ah, I think I understand... Since every mail we send is personalised ("Dear Foo") the chunks will only ever have size 1, correct? 

I guess it's a good idea then to put a comment in `settings.py`.

Reply:
No, that comes from using `djcelery-mail` and has nothing to do with the way we address our mails.

To send mails, we create a list of mail data like subject, message, sender, recipient and so on and pass it to `send_mass_mail()`. This list can contain as many mails, as we want. `djcelery-mail` then splits this list in chunks of `CELERY_EMAIL_CHUNK_SIZE` emails per chunk. It then tries to send those mails and we get a `AsyncResult` for every chunk of mails.

The default value of `CELERY_EMAIL_CHUNK_SIZE` is 10. So lets say we have a chunk of 10 mails and 2 of them produce errors while sending. `send_mass_mail()` will give us one `AsyncResult` for this chunk, which in turn will give us the number of mails actually sent. In this case, this result will be 8, so we know that 8 out of 10 mails were sent, but we don't know which 8 or more importantly which 2 failed. `djcelery-mail` produces a log, which we could pars and see which mails have failed, but by setting `CELERY_EMAIL_CHUNK_SIZE` to 1, we don't need that. Instead we get a `AsyncResult` for every mail and can see what exactly worked and what didn't.

If now somebody would set `CELERY_EMAIL_CHUNK_SIZE` to anything but 1, `zip(mail_list, send_results)` would misalign the mails with the results and set status incorrectly.

This PR introduces a way to set local settings in separate file anyway, so I think it should be enough to add a note to README.md, to not change any settings in settings.py, but in local_settings.py and never to change `CELERY_EMAIL_CHUNK_SIZE`.

Reply:
I see, sounds like a major footgun, though. Add the readme entry and open an issue so that we don't forget to at some point do something against the `zip` and then we're good to go.